pkg/kubelet/dockershim/docker_container.go

```diff
@@ -0,0 +1,196 @@
+	}
+	if sandboxConfig == nil {
+		return "", fmt.Errorf("sandbox config is nil for container %q", config.GetName())
+	}
+
+	// Merge annotations and labels because docker supports only labels.
+	// TODO: should we apply docker-specific labels?
+	labels := config.GetLabels()
+	for k, v := range config.GetAnnotations() {
+		if _, ok := labels[k]; !ok {
+			// Only write to labels if the key doesn't exist.
+			labels[k] = v
+		}
+	}
+
+	createConfig := dockertypes.ContainerCreateConfig{
+		Name: config.GetName(),
+		Config: &dockercontainer.Config{
+			// TODO: set User.
+			Hostname:   sandboxConfig.GetHostname(),
+			Entrypoint: dockerstrslice.StrSlice(config.GetCommand()),
+			Cmd:        dockerstrslice.StrSlice(config.GetArgs()),
+			Env:        generateEnvList(config.GetEnvs()),
+			Image:      config.GetImage().GetImage(),
+			WorkingDir: config.GetWorkingDir(),
+			Labels:     labels,
+			// Interactive containers:
+			OpenStdin: config.GetStdin(),
+			StdinOnce: config.GetStdinOnce(),
+			Tty:       config.GetTty(),
+		},
+	}
+
+	// Fill the HostConfig.
+	hc := &dockercontainer.HostConfig{
+		Binds:          generateMountBindings(config.GetMounts()),
+		ReadonlyRootfs: config.GetReadonlyRootfs(),
+		Privileged:     config.GetPrivileged(),
+	}
+
+	// Apply options derived from the sandbox config.
+	if lc := sandboxConfig.GetLinux(); lc != nil {
+		// Apply Cgroup options.
+		// TODO: Check if this works with per-pod cgroups.
+		hc.CgroupParent = lc.GetCgroupParent()
+
+		// Apply namespace options.
+		sandboxNSMode := fmt.Sprintf("container:%v", podSandBoxID)
+		hc.NetworkMode = dockercontainer.NetworkMode(sandboxNSMode)
+		hc.IpcMode = dockercontainer.IpcMode(sandboxNSMode)
+		hc.UTSMode = ""
+		hc.PidMode = ""
+
+		nsOpts := lc.GetNamespaceOptions()
+		if nsOpts != nil {
+			if nsOpts.GetHostNetwork() {
+				hc.UTSMode = namespaceModeHost
+			}
+			if nsOpts.GetHostPid() {
+				hc.PidMode = namespaceModeHost
+			}
+		}
+	}
+
+	// Apply Linux-specific options if applicable.
+	if lc := config.GetLinux(); lc != nil {
+		// Apply resource options.
+		// TODO: Check if the units are correct.
+		// TODO: Can we assume the defaults are sane?
+		rOpts := lc.GetResources()
+		if rOpts != nil {
+			hc.Resources = dockercontainer.Resources{
+				Memory:     rOpts.GetMemoryLimitInBytes(),
+				MemorySwap: -1, // Always disable memory swap.
+				CPUShares:  rOpts.GetCpuShares(),
+				CPUQuota:   rOpts.GetCpuQuota(),
+				CPUPeriod:  rOpts.GetCpuPeriod(),
+				// TODO: Need to set devices.
+			}
+			hc.OomScoreAdj = int(rOpts.GetOomScoreAdj())
+		}
+		// Note: ShmSize is handled in kube_docker_client.go
+	}
+
+	// TODO: Seccomp support. Need to figure out how to pass seccomp options
+	// through the runtime API (annotations?).See dockerManager.getSecurityOpts()
+	// for the details. Always set the default seccomp profile for now.
+	hc.SecurityOpt = []string{fmt.Sprintf("%s=%s", "seccomp", defaultSeccompProfile)}
+
```

Comment:
I can't find where we implement that in `dockertools`, hence I skipped it. Added a TODO to check later.